### DISCUSSION TITLE
HTTP 429 Error

### DISCUSSION THREAD

**Discussion by meleila** [microsoft-search/pnp-modern-search#3178] [Q&A] [answered]
Hi,
we have a problem with loading the pnp search results web part. Each time a user opens the site for the first time the web part shows the HTTP 429 error. After a reload everything works fine.

Therefore, the error is difficult to reproduce, resetting cache doesnt reproduce the issue again. We can only ask users who never called the site to test.

If we open the issue in the network tab of the dev tool the following error shows: "-1, Microsoft.Office.Server.Search.REST.SearchServiceException"

Modern search version: 4.8
Browser: Edge

**Comment by kasperbolarsen** [comment]
If possible please upgrade to the current release version, as we can not support prior versions. 
Does this issue only exists on a specific tenant or on multiple tenants?

And please show which data source you are using and the query.

**Comment by wobba** [comment]
429 is throttling at the API, and is just that. Which means there are too many requests being made. We might can improve and handle 429's on retry, if you share more which API call is being throttled.

**Reply by meleila** [reply]
Hi, we updated to version 4.9.1 but the error still persists. 
The request failing is "...sharepoint.com/_api/search/postquery". We use the WebPart 4 times on the site (also chrashed with only 2 times) and filter for content type and content class, RowLimit is 5.
Strange is still that it only crashes when opening the first time, but like this we still can't go live.
Thx :)

**Reply by wobba** [reply]
You might have other solutions using search running in your portals? How many users in the tenant? As all API requests are throttled as one. See https://learn.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online for more information on throttling issues and SharePoint API's.

Thus the version of the web parts does not matter - it's all about the number og API requests happening. I'll take a look and see if we can add handling of it, which means the search web parts would look like they are loading until they are allowed to fetch data.

**Reply by anditee** [reply]
Hey, my colleque @meleila is out of office at this time so I will provide the information you asked for.

1. This is the first solution we used the pnpSearch Webpart in this tenant
2. There are about 500 people on this tenant, but the quantity of user doesn't seem to be the problem. On every user the first time load of the page results in the error 429. of the webpart. After refreshing everything works fine again.
3. Maybe the problem is that we are using the same webpart 4 times on the current site this is happening. But we tried this in another tenant and there was no problem. Further more we used several Result webparts on different tenants and never had such issues with throttling. (Some of them are even searching in about 10000 files and they work like a charm.)

**Reply by kasperbolarsen** [reply]
Hi
I have had similar issues on some of my customers tenants, and the reason was that the tenant was in a farm that was under a heavy load. We had to raise a support ticket with microsoft and prove that we were throttled when doing trivial calls. 
Each time the problem "disappeared by itself" after about a week.....
So I would suggest raising a support ticket med MS Support.

**Reply by anditee** [reply]
@kasperbolarsen Hi, thanks for the quick response. We are fine if this is a problem on Microsofts side and we are willing to contact the microsoft support. But shouldn't there be a error / throttling handling on the Webparts side? And if there is one why doesn't it help in this specific case.

Biggest problem is that Microsoft will refere to 'that is not a microsoft sided problem and the developers on the other side should fix it' we had similar issues and always didn't get any support because of 3rd party development as soon we reached out to microsoft.

Thanks in advance for future answers.

**Comment by kasperbolarsen** [comment]
Hi 

Before calling MS you should be able to verify the issue in Classic or in the API. I suggest running a query at [yourtenant]/search/Pages/results.aspx?k=test  as that should show the same error.

**Reply by anditee** [reply]
Thanks for the tipp. I executed the search and there was no problem. Statuscode = 200

**Reply by wobba** [reply]
If you get 200, I would expect the web part to work for you as well... can you try it late one evening or on the weekend, as throttling is less likely to occur.

**Reply by meleila** [reply]
Hi, we did try in the evening, unfortunately for one of our testers (not for the other) the 429 error still occured in one of the two WebParts on the site. Could handling throttling solve the issue? thx

**Reply by wobba** [reply]
We likely need to update pnpjs to get better throttling handling so may take some time.

**Reply by anditee** [reply]
@wobba hey the project was on a halt for a while and we are still facing the same issues. Is there any kind of information about when the solution with extended throttling is going to be published? Within the next days i will update the search webpart on our side and test again if a recent update did fix the issue.

**Reply by wobba** [reply]
No timelines, and depends on spare time if I'm to look into it.

**Reply by anditee** [reply]
Is there anyway we can support you in any regards so this might be fixed earlier? We can offer manpower and try to dig into the problem our self and try to find a suitable solution to this.

Best regards

**Reply by anditee** [reply]
@wobba Hi,

We have contacted microsoft regarding the problem. According to microsoft, the problem is not with the webpart per se, but rather that the webpart is loaded last by the tenant. However, no other actions are performed by the tenant. So the initial start of the webpart fails and causes the error. Is there a way to fix this with a quickfix? Microsoft itself does not offer any support to find the real reason for this problem.

Best regards

**Reply by wobba** [reply]
429 are throttling errors, and throttling is across users/API's etc.. so hard to know why it happens and on what call. You can check the network tab in the browser and see exactly what call is getting 429.

**Reply by FloWaeger** [reply]
We have these problems right now in different tenants. Are there any updates regarding the idea to improve pnp web part to load when it is allowed to fetch data?

**Reply by kasperbolarsen** [reply]
Do you see the same issue at [yourtenant]/search/Pages/results.aspx?k=test ?

**Reply by FloWaeger** [reply]
Not the same. There is a red banner.

<img width="1438" height="86" alt="image" src="https://github.com/user-attachments/assets/bf83b603-c9d1-444f-addf-221d5cad67ea" />

These are the errors on Intranet start page:

<img width="684" height="229" alt="image" src="https://github.com/user-attachments/assets/2d7cd8c6-de1a-41da-a91f-77106f54a567" />

**Reply by kasperbolarsen** [reply, marked as answer]
So the Search API is failing on the classic search portal, ergo is this a Microsoft issue, and you should create a service request, using the classic search portal as "evidence". 
There is nothing we can do in the PnP project as we are just using the same API
